client/js/store/modules/auth.js

```diff
@@ -107,5 +107,51 @@ export default {
         );
       });
     },
+    async requestStatus(context) {
+      if (!window) {
+        // Swallow error, this call is only supported client side
+        return;
+      }
+
+      const { location } = window;
+      const isLobby = location.hash.startsWith('#/lobby');
+      const isGame = location.hash.startsWith('#/game');
```

Comment:
So I'm fuzzy on the bootstrapping lifecycle. I'm surprised the vue router won't necessarily be available at this time because we're inside a vuex action so doesn't the app have to have mounted for this function to fire? Shouldn't that guarantee that the vue router is available? Not that this is a terrible way to check the route, I'd just like to learn more about when which resources are available and why this was necessary

Reply:
![image](https://user-images.githubusercontent.com/414475/187927107-37e322a2-8323-4f73-bd1b-8f88ba23f260.png)

This is what happens if you try and pass the router to this action before we's mounted vue. I have a feeling we could potentially a lifecycle method the app itself and fire stuff there with the router but in my initial attempts I was not successful due to the order of operations. Something like this:
```
const app = new Vue({
  vuetify,
  store,
  router,
  async beforeCreate() {
    // Make sure we try and reestablish a player's session if one exists
    // We do this before the app mounts to preempt the reauth/logout logic
    try {
      await store.dispatch('requestStatus', this.$route);
    } catch {
      // Unable to validate player status or player not authenticated
      // Swallow the error to let the player continue
    }
  },
  render: (h) => h(App),
}).$mount('#app');
```

But what happens is that the vue router beforeResolve fires before the beforeCreate lifecycle method which will redirect you to the /login route when you're actually authed. Something to look into as a follow up, though, for sure.

Reply:
The other thing we could potentially look in to doing is adding this call to vue router middleware, but I'd need to figure out how to run this once before the router did everything else.